doc/faq/troubleshooting_faq.rst

```diff
@@ -114,35 +114,17 @@ provide the following information in your e-mail to the `mailing list
   the error will help you find a bug in *your* code that is causing the
   problem.
 
-* You can get helpful debugging output from Matlotlib by using the `logging`
-  library in your code and posting the verbose output to the lists.  For a
-  command-line version of this, try::
+* Matplotlib provides debugging information through the `logging` library, and
+  a helper function to set the logging level: one can call ::
 
-     python -c "from logging import *; basicConfig(level=DEBUG); from pylab import *; plot(); show()"
+    plt.set_loglevel("info")  # or "debug" for more info
 
+  to obtain this debugging information.
 
-  If you want to put the debugging hooks in your own code, then the
-  most simple way to do so is to insert the following *before* any calls
-  to ``import matplotlib``::
-
-    import logging
-    logging.basicConfig(level=logging.DEBUG)
-
-    import matplotlib.pyplot as plt
-
-  Note that if you want to use `logging` in your own code, but do not
-  want verbose Matplotlib output, you can set the logging level
-  for Matplotlib independently::
-
-    import logging
-    # set DEBUG for everything
-    logging.basicConfig(level=logging.DEBUG)
-    logger = logging.getLogger('matplotlib')
-    # set WARNING for Matplotlib
-    logger.setLevel(logging.WARNING)
-
-  The `logging` module is very flexible, and can be a valuable tool in chasing
-  down errors.
+  (Standard functions from the `logging` module are also applicable; e.g. one
+  could call ``logging.basicConfig(level="DEBUG")`` even before importing
+  Matplotlib, or attach a custom handler to the "matplotlib" logger.  This may
+  be useful if you use a custom logging configuration.)
```

Comment:
This needs a note as well that sometimes logging before the first import is necessary in order to get the startup information.  i.e.

```
If debug information is needed during the initial Matplotlib imports, the logging module should be configured before the first Matplotlib import by adding "import logging; logging.basicConfig(level="DEBUG")" to the top of your script.
```

Reply:
Added that to the parenthetical note.